packages/twenty-front/src/modules/apollo/optimistic-effect/utils/triggerCreateRecordsOptimisticEffect.ts

```diff
@@ -19,28 +21,53 @@ import { parseApolloStoreFieldName } from '~/utils/parseApolloStoreFieldName';
   We need to refactor how the record creation works in the RecordTable so the created record row is temporarily displayed with a local state,
   then we'll be able to uncomment the code below so the cached lists are updated coherently with the variables.
 */
+type TriggerCreateRecordsOptimisticEffectArgs = {
+  cache: ApolloCache<object>; //oof type signature might be necessary to provide Generic
+  objectMetadataItem: ObjectMetadataItem;
+  recordsToCreate: RecordGqlNode[];
+  objectMetadataItems: ObjectMetadataItem[];
+  shouldMatchRootQueryFilter?: boolean;
+  checkForRecordInCache?: boolean;
+};
 export const triggerCreateRecordsOptimisticEffect = ({
   cache,
   objectMetadataItem,
   recordsToCreate,
   objectMetadataItems,
   shouldMatchRootQueryFilter,
-}: {
-  cache: ApolloCache<unknown>;
-  objectMetadataItem: ObjectMetadataItem;
-  recordsToCreate: RecordGqlNode[];
-  objectMetadataItems: ObjectMetadataItem[];
-  shouldMatchRootQueryFilter?: boolean;
-}) => {
-  recordsToCreate.forEach((record) =>
+  checkForRecordInCache,
+}: TriggerCreateRecordsOptimisticEffectArgs) => {
+  const getCurrentSourceRecord = (recordId: string): RecordGqlNode | null => {
+    if (checkForRecordInCache === true) {
```

Comment:
let's add a default in the signature for checkForRecordInCache = true

Reply:
let's simplify as much as possible our functions signature and not call them if not needed:
`const getCurrentSourceRecord = (recordId: string): RecordGqlNode =`

Reply:
I think the most common use case would be not to check in the cache ?
From my understanding we should be cache checking only on optimistic update on an api response